libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -632,6 +632,73 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
         return (await client.userToken.getAadTokens(userId, connectionName, { resourceUrls: resourceUrls }, { channelId: context.activity.channelId }))._response.parsedBody as {[propertyName: string]: TokenResponse };
     }
 
+    /**
+     * Asynchronously Get the raw signin resource to be sent to the user for signin.
+     * 
+     * @param context The context object for the turn.
+     * @param connectionName The name of the auth connection to use.
+     * @param userId The user id that will be associated with the token.
+     * @param finalRedirect The final URL that the OAuth flow will redirect to.
+     * 
+     * @returns The [BotSignInGetSignInResourceResponse](xref:botframework-connector.BotSignInGetSignInResourceResponse) object.
+     */
+    public async getSignInResource(context: TurnContext, connectionName: string, userId?: string, finalRedirect?: string): Promise<TokenApiModels.BotSignInGetSignInResourceResponse>
+    {
+        if (!connectionName) {
+            throw new Error('getUserToken() requires a connectionName but none was provided.');
+        }
+
+        if (!context.activity.from || !context.activity.from.id) {
+            throw new Error(`BotFrameworkAdapter.getSignInResource(): missing from or from.id`);
+        }
+
+        // what to do when userId is null (same for finalRedirect)
+        if(userId && context.activity.from.id != userId) {
+            throw new Error('BotFrameworkAdapter.getSiginInResource(): cannot get signin resource for a user that is different from the conversation');
+        }
+
+        const url: string = this.oauthApiUrl(context);
+        const client: TokenApiClient = this.createTokenApiClient(url);
+        const conversation: Partial<ConversationReference> = TurnContext.getConversationReference(context.activity);
+
+        const state: any = {
+            ConnectionName: connectionName,
+            Conversation: conversation,
+            //RelatesTo = context.activity.RelatesTo,
```

Comment:
Remove comment

Reply:
Note that C# PR fills the RelatesTo with the root conversation Id in the case of skills. It's one of the latest check ins from Jeff in C#. Please update that change as it's pretty important

Reply:
added in latest commit